include/seastar/core/shared_ptr_debug_helper.hh

```diff
@@ -60,7 +65,9 @@ public:
     }
 private:
     void check() const {
-        assert(_cpu == std::this_thread::get_id());
+        if (__builtin_expect(_cpu != std::this_thread::get_id(), false)) {
```

Comment:
I think C++20 `[[unlikely]]` is nicer, but this is also correct.

Reply:
agreed, but we still have to support c++17 in seastar